src/CacheCow.Client.FileCacheStore/FileStore.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using CacheCow.Common;
+
+namespace CacheCow.Client.FileCacheStore
+{
+    /// <summary>
+    /// A simple 'cache-to-file' storage with persistanty over multiple runs.
+    /// </summary>
+    public class FileStore : ICacheStore
+    {
+        private readonly MessageContentHttpMessageSerializer _serializer = new MessageContentHttpMessageSerializer();
+
+        private readonly string _cacheRoot;
+
+        /// <summary>
+        /// Minimum expiry of items. Default is 6 hours.
+        /// Bear in mind, even expired items can be used if we do a cache validation request and get back 304
+        /// </summary>
+        public TimeSpan MinExpiry { get; set; }
+
+
+        /// <inheritdoc />
+        public FileStore(string cacheRoot)
+        {
+            _cacheRoot = cacheRoot;
+            if (!Directory.Exists(_cacheRoot))
+            {
+                Directory.CreateDirectory(cacheRoot);
+            }
+        }
+
+        /// <inheritdoc />
+        public async Task<HttpResponseMessage> GetValueAsync(CacheKey key)
+        {
+            if (!File.Exists(_pathFor(key)))
+            {
+                return null;
+            }
+
+            var fs = File.OpenRead(_pathFor(key));
```

Comment:
Probably should wrap this in a `try` / `finally` or a `using` to ensure that if the `DeserializeToResponseAsync` throws we always Close/Dispose the file stream.